eyepop/worker/worker_endpoint.py

```
@@ -582,13 +582,19 @@ async def _pipeline_request_with_retry(self, method: str, url_path_and_query: st
-                            response = await self.client_session.request(method, url, headers=headers,
-                                                                         data=data.getvalue(), timeout=timeout)
-                        else:
-                            response = await self.client_session.request(method, url, headers=headers, data=data,
-                                                                         timeout=timeout)
+                    data = open_data()
+                    if isinstance(data, StringIO):
+                        response = await self.client_session.request(method, url, headers=headers,
+                                                                     data=data.getvalue(), timeout=timeout)
+                    elif isinstance(data, AsyncIterable):
+                        async def data_streamer():
+                            async for chunk in data:
```

Ruff (0.15.18)

[warning] Function definition does not bind loop variable `data`

(B023)